src/java/voldemort/server/protocol/admin/AdminServiceRequestHandler.java

```diff
@@ -1131,12 +1135,18 @@ public void operate() {
 
                             String destinationDir = store.getStoreDirPath() + File.separator + "version-"
                                             + Long.toString(pushVersion);
+
+                            store.setCurrentlyFetching(true);
```

Comment:
This is prone to thread synchronization problems. There are also more corner cases where the job is waiting to be fetched in the queue, then concurrent pushes can happen as well.